### ISSUE TITLE
Split linter to there modules `core`, `rails` and `rspec`

### ISSUE THREAD

**Issue by roman-dubrovsky** [roman-dubrovsky/datarockets-style#2]
The main idea is adding separate modules for some specific rules: 

`core` - which we're using for Ruby apps and gems (exclude rails specific rules)
`rails` - which we're using for Rails apps
`rspec` - which we're using for rspec tests

**Comment by AleksSenkou** [comment]
@roman-dubrovsky can "rails" be different to "rails-api"? Looks like pretty the same, but maybe

**Comment by roman-dubrovsky** [comment]
@AleksSenkou I don't think so. For rails api we're using the same subjects (excluding templates but rubocop does not check template)

**Comment by roman-dubrovsky** [comment]
For this issue, I'm going:

* [ ] add a separate config for rspecs 
* [ ] rename `default` config to `base` 
* [ ] add `ruby` and `rails` configs. For now it should contains only `base ` config + different exclusions 
* [ ] move style guides into the `doc` directory
* [ ] add separate files for Rails and Rspec style guides
* [ ] add links to that style guides to the main style guides
* [ ] Updates Readme and describes files which we have 
* [ ] Update Change log file
